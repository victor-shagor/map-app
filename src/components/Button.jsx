import React from "react";
import ClipLoader from "react-spinners/ClipLoader";

const Button = ({ text, onClick, isLoading = false }) => {
  return (
    <button
      type="submit"
      onClick={onClick}
      disabled={isLoading}
      className="btn"
    >
      {!isLoading ? text : <ClipLoader color="white" size={35} />}
    </button>
  );
};

export default Button;
