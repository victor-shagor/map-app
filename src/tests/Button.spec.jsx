import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import Button from "components/Button";

describe("Button Component", () => {
  it("render button component", () => {
    const onClick = jest.fn();

    render(<Button text={"test"} onClick={onClick} />);
    expect(screen.getByRole("button")).toHaveTextContent("test");
  });

  it("onclick event should be triggered", () => {
    const onClick = jest.fn();

    render(<Button text={"test"} onClick={onClick} />);
    const buttonClick = screen.getByRole("button");
    expect(onClick).toHaveBeenCalledTimes(0);
    fireEvent.click(buttonClick);
    expect(onClick).toHaveBeenCalled();
  });

  it("renders correctly", () => {
    const onClick = jest.fn();
    const tree = renderer
      .create(<Button text={"test"} onClick={onClick} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
