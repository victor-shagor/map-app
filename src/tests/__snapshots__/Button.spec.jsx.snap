// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
<button
  className="btn"
  disabled={false}
  onClick={[MockFunction]}
  type="submit"
>
  test
</button>
`;
